.github/workflows/deploy-smart-contracts.yml

```
@@ -17,12 +17,27 @@ on:
         required: true
         type: choice
         options:
-          - develop
-          - production
-        default: 'develop'
+          - testnets
+          - mainnets
+        default: 'testnets'
 
 jobs:
+  validate:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Validate mainnet deployment conditions
+        run: |
+          if [[ "${{ inputs.environment }}" == "mainnets" && "${{ inputs.network }}" == "bellecour" && "${{ github.ref }}" != "refs/heads/main" ]]; then
```

```suggestion
          if [[ "${{ inputs.environment }}" == "mainnets" && !("${{ inputs.network }}" == "bellecour" && "${{ github.ref }}" == "refs/heads/main") ]]; then
```
To make sure we only use `mainnets` with Bellecour.

---

I'm not sure, since mainnets could be the mainnets of arbitrum or any other mainnets live (?)

---

For this step, I guess yes it's the option but I should add a todo for later :)